### PR TITLE
CORE-12407: Fix lingering OSGi issues in the worker applications.

### PR DESCRIPTION
**Pull request by chrisr3**
- Identify OSGi fragment bundles via `BundleRevision.getTypes()` flags rather than by searching bundle headers.
- Fix `loadClass` and `getResources` methods for `OsgiClassLoader` to support multiple bundles correctly.
- Ensure uncaught lifecycle exceptions are logged properly instead of just printed to console.
- Replace `OsgiDelegatedClassLoader` with `BundleWiring.getClassLoader()`, and delete `OsgiDelegatedClassLoader`.

### CODE REVIEW

File: libs/lifecycle/lifecycle-impl/src/main/kotlin/net/corda/lifecycle/impl/LifecycleCoordinatorSchedulerFactoryImpl.kt [hunk 1]
```diff
@@ -7,9 +7,16 @@ import net.corda.lifecycle.LifecycleCoordinatorSchedulerFactory
 import org.osgi.service.component.annotations.Activate
 import org.osgi.service.component.annotations.Component
 import org.osgi.service.component.annotations.Deactivate
+import org.slf4j.LoggerFactory
 
 @Component(service = [ LifecycleCoordinatorSchedulerFactory::class ])
 class LifecycleCoordinatorSchedulerFactoryImpl @Activate constructor(): LifecycleCoordinatorSchedulerFactory {
+    private val logger = LoggerFactory.getLogger(this::class.java)
+
+    private fun handleUncaughtException(thread: Thread, exception: Throwable) {
```
Comment (mnesbit) [inline libs/lifecycle/lifecycle-impl/src/main/kotlin/net/corda/lifecycle/impl/LifecycleCoordinatorSchedulerFactoryImpl.kt:16]: Any chance we can get the lifecycle object name?

Comment (chrisr3) [reply libs/lifecycle/lifecycle-impl/src/main/kotlin/net/corda/lifecycle/impl/LifecycleCoordinatorSchedulerFactoryImpl.kt:16]: It's the "uncaught exception handler", so where would I get that from? All I have to work with is the thread and the uncaught exception, and it's better for that to get logged properly than just sent to the console. I'm hoping you can infer the context from the thread name, and anything else in the Log4J logger context. And from the exception stack trace, of course.

So short answer: "No".

**Review by mnesbit — APPROVED** [review]
OK

File: libs/lifecycle/lifecycle-impl/src/main/kotlin/net/corda/lifecycle/impl/LifecycleCoordinatorSchedulerFactoryImpl.kt (hunk 1 above)
Comment (mnesbit) [reply libs/lifecycle/lifecycle-impl/src/main/kotlin/net/corda/lifecycle/impl/LifecycleCoordinatorSchedulerFactoryImpl.kt:16]: The task that was being executed ideally

Comment (chrisr3) [reply libs/lifecycle/lifecycle-impl/src/main/kotlin/net/corda/lifecycle/impl/LifecycleCoordinatorSchedulerFactoryImpl.kt:16]: > The task that was being executed ideally

If the "task name" (`LifecycleCoordinator.name`?) is contained in either the logger context, the thread or the exception then it's logged already. But I cannot change the definition of what the [UncaughtExceptionHandler](https://docs.oracle.com/javase/7/docs/api/java/lang/Thread.UncaughtExceptionHandler.html) is.

However, considering that the default ultimate `UncaughtExceptionHandler` is effectively:
```java
if (!(e instanceof ThreadDeath)) {
    System.err.print("Exception in thread \"" + t.getName() + "\" ");
    e.printStackTrace(System.err);
}
```
I still think this is an improvement.

**Review by driessamyn — APPROVED** [review]
LGTM

**Pull request merged**